pulser-core/pulser/sequence/sequence.py

```diff
@@ -2118,10 +2128,19 @@ def _check_qubits_give_ids(
         return ids
 
     @seq_decorators.block_if_measured
-    def _delay(self, duration: Union[int, Parametrized], channel: str) -> None:
+    def _delay(
+        self,
+        duration: Union[int, Parametrized],
+        channel: str,
+        at_rest: bool = False,
+    ) -> None:
         self._validate_channel(channel, block_if_slm=True)
         if self.is_parametrized():
             return
+        if at_rest:
+            self._schedule.wait_for_fall(channel)
+        if not duration:
+            return
```

Comment:
So one can ask for a delay of duration 0 but with at_rest=True and it will create a delay until last pulse reaches 0 amplitude and detuning ?

Reply:
I am fine with that behaviour :)

Reply:
Tested in test_sequence.py